### PR TITLE
feat: support openjdk-jre

### PR DESCRIPTION
**Pull request by karniwl**
- [x] Ready for review
- [x] Follows CONTRIBUTING rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
Adds support for detecting `openjdk-jre` binary.

#### How should this be manually tested?
Test `snyk test --docker openjdk:latest`.

### CODE REVIEW

File: lib/analyzer/binary-version-extractors/node.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+export {
+  extract,
+  packageNames,
+};
+
+const packageNames = ['node', 'nodejs'];
+
+function extract(targetImage: string)
+  : Promise<Binary | null> | null {
+  return new Docker(targetImage).run('node', [ '--version' ])
```
Comment (darmalovan) [inline lib/analyzer/binary-version-extractors/node.ts]: Can we asyncify everything here?

File: lib/analyzer/binary-version-extractors/openjdk-jre.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  Java(TM) SE Runtime Environment (build 1.8.0_191-b12)
+  Java HotSpot(TM) 64-Bit Server VM (build 25.191-b12, mixed mode)`
+  extracting `1.8.0_191-b12`
+  */
+  if (!fullVersionOutput) {
+    return null;
+  }
+  const bracketsRE = /\(build (.*)\)$/;
+  const jdkVersionLines = fullVersionOutput.split('\n');
+  const buildVersion = jdkVersionLines && jdkVersionLines.length > 1 &&
```
Comment (darmalovan) [inline lib/analyzer/binary-version-extractors/openjdk-jre.ts]: I might be even stricter and say:
 if (jdkVersionLines.length) !== 3 {
   return null;
}

File: lib/analyzer/binary-version-extractors/openjdk-jre.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  extract,
+  packageNames,
+};
+
+// todo: add common names
+const packageNames = ['java'];
+
+function extract(targetImage: string)
+  : Promise<Binary | null> | null {
+  return new Docker(targetImage).run('java', [ '-version' ])
```
Comment (darmalovan) [inline lib/analyzer/binary-version-extractors/openjdk-jre.ts]: Can async here as well?
I thought maybe to extract the new Docker logic as well, but maybe let's wait for to the next binary.

File: lib/analyzer/binary-version-extractors/openjdk-jre.ts (hunk 2 above)
Comment (karniwl) [reply lib/analyzer/binary-version-extractors/openjdk-jre.ts]: for some reason the length is 4 but I did adjust it, let me know what you think

File: test/lib/analyzer/index.test.ts [hunk 4]
```diff
@@ -82,6 +82,24 @@ test('analyzer', async t => {
...
+      '-version',
+    ])
+      .callsFake(async (docker,
+        [run, rm, entry, empty, network, none, image]) => {
+        try {
+          const example = examples[image];
+          return readOsFixtureFile(example.dir, 'openjdk-jre-version.txt');
+        } catch {
+          // tslint:disable-next-line:no-string-throw
+          throw 'docker: Error running `docker java --version`';
```
Comment (darmalovan) [inline test/lib/analyzer/index.test.ts]: on dash `-version`

**Pull request merged**